openquake/hazardlib/tests/sourceconverter_test.py

```diff
@@ -85,14 +85,89 @@
+                </occurRates>
+                <lengths>
+                    25
+                </lengths>
+            </multiMFD>
+            <nodalPlaneDist>
+                <nodalPlane dip="9.0000000E+01" probability="1.0000000E+00" rake="0.0000000E+00" strike="2.7000000E+02"/>
+            </nodalPlaneDist>
+            <hypoDepthDist>
+                <hypoDepth depth="5.0000000E+00" probability="7.0000000E-02"/>
+                <hypoDepth depth="1.5000000E+01" probability="2.4000000E-01"/>
+                <hypoDepth depth="2.5000000E+01" probability="1.2000000E-01"/>
+                <hypoDepth depth="3.5000000E+01" probability="5.7000000E-01"/>
+            </hypoDepthDist>
+        </multiPointSource>
+    </sourceGroup>
+</nrml>
+'''
+
 
 class PointToMultiPointTestCase(unittest.TestCase):
-    def test(self):
+    def test_simple(self):
         testfile = os.path.join(testdir, 'two-point-sources.xml')
         sm = nrml.read(testfile).sourceModel
-        update_source_model(sm)
+        update_source_model(sm, testfile)
         with io.BytesIO() as f:
             nrml.write(sm, f)
             got = f.getvalue().decode('utf-8')
             print(got)
             self.assertEqual(got, expected)
+
+    def test_complex(self):
+        testfile = os.path.normpath(os.path.join(
+            testdir, '../../../qa_tests_data/classical/case_30/ssm/shallow/'
+            'gridded_seismicity_source_4.xml'))
+        sm = nrml.read(testfile).sourceModel
+        update_source_model(sm, testfile)
+        with io.BytesIO() as f:
+            nrml.write(sm, f)
+            got = f.getvalue().decode('utf-8')
+            print(got)
```

Comment:
Should we remove this print?

Reply:
No, it is there for debugging purpose. Notice that since the tests are run with nose, print are hidden unless the `-s` option is given, which sometimes I do since that print is useful.